src/main/java/services/RestoreFactory.java

```diff
@@ -443,34 +445,54 @@ private String getSyncDeviceId() {
     }
 
     public HttpHeaders getUserHeaders() {
+        if (getHqAuth() == null) {
+            throw new RuntimeException("Trying to get Authentication headers but request " +
+                    " did not have an authentication key.");
+        }
         HttpHeaders headers = getHqAuth().getAuthHeaders();
         headers.set("X-CommCareHQ-LastSyncToken", getSyncToken());
         headers.set("X-OpenRosa-Version", "3.0");
         headers.set("X-OpenRosa-DeviceId", getSyncDeviceId());
         return headers;
     }
 
-    public String getRestoreUrl() {
+    public Pair<String, HttpHeaders> getRestoreUrlAndHeaders() {
         if (caseId != null) {
```

Comment:
we should make sure that the request to formplayer was validated with mac digest validation before we allow using the case restore endpoint. does that happen further up or should that happen here?

Reply:
Also happens here https://github.com/dimagi/formplayer/blob/2fde9891f8bc8ea047c140447ffaff7d57740a8a/src/main/java/application/FormplayerAuthFilter.java#L56-L56